src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+
+        for (const key of action.keysToRemove) {
+          if (state[key]) {
+            delete state[key];
+          }
+        }
+        break;
+      case 'clear':
+
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default: return null;
+    }
+  }
+
+  return state;
```

Comment:
maybe you dont need to return state

Reply:
idk
but I thought that we should to return transformed state

Reply:
nah :D
no return statement required, let's remove it
![Screenshot 2023-03-16 at 09 57 48](https://user-images.githubusercontent.com/75496546/225551462-9db76b82-7fef-4e74-a748-cf24e95034f9.png)